### PR TITLE
[Core] Cleanup indexed object

### PR DESCRIPTION
**Pull request by loumalouomega**
**Description**
I was looking to something and I realized that indexed object wasn't at includes. This cleans up indexed object and it moves from utilities to includes (as it should)

**Changelog**
- Cleans up indexed object 
- Moves from utilities to includes

### CODE REVIEW

File: kratos/includes/indexed_object.h [hunk 1]
```diff
@@ -166,69 +153,54 @@ class IndexedObject
...
     ///@{
 
-
     ///@}
     ///@name Member Variables
     ///@{
 
-    IndexType mId;
-
+    IndexType mId; /// The Id hold by th object
```
Comment (RiccardoRossi) [inline kratos/includes/indexed_object.h]: held

**Review by RiccardoRossi — COMMENTED** [review]
very minor comment

**Comment by roigcarlo** [comment]
Please double check that this compiles in win and linux **without cotire**. Not saying is wrong but the CI does not detect errors caused by these changes, so extra care.

File: kratos/includes/indexed_object.h (hunk 1 above)
Comment (loumalouomega) [inline kratos/includes/indexed_object.h]:
```suggestion
    IndexType mId; /// The Id held by the object
```

**Comment by loumalouomega** [comment]
> Please double check that this compiles in win and linux **without cotire**. Not saying is wrong but the CI does not detect errors caused by these changes, so extra care.

I compiled without cotire : D

**Comment by maceligueta** [comment]
Should we leave a deprecated file in ` kratos/utilities/indexed_object.h` with a compilation pragma saying that this file is no longer valid? I am afraid that people in forks or independent apps will get a bit lost with this.

**Comment by loumalouomega** [comment]
> Should we leave a deprecated file in ` kratos/utilities/indexed_object.h` with a compilation pragma saying that this file is no longer valid? I am afraid that people in forks or independent apps will get a bit lost with this.

perhaps

**Comment by philbucher** [comment]
> Should we leave a deprecated file in ` kratos/utilities/indexed_object.h` with a compilation pragma saying that this file is no longer valid? I am afraid that people in forks or independent apps will get a bit lost with this.

I was thinking the same, this is how boost does it afaik
=> `#pragma message` I think is supported by most compilers

**Review by philbucher — COMMENTED** [review]
nice!

**Comment by maceligueta** [comment]
Awesome, good PR!

**Review by philbucher — APPROVED** [review]
are there tests for this?
I think not right?

=> it would be awesome if you could add some, esp the copy and assignment ctors and the serializer would be good to be tested

Up to you, nice work in any case, thx

**Pull request merged**